index.bs

```diff
@@ -47,66 +51,88 @@ dictionary CSSStyleSheetInit {
 };
 </pre>
 
+
 <dl>
-	<dt><dfn method for=Document lt="createCSSStyleSheet(text)|createCSSStyleSheet(text, options)">createCSSStyleSheet(text, options)</dfn></dt>
+	<dt><dfn constructor for=CSSStyleSheet lt="CSSStyleSheet()|CSSStyleSheet(options)">CSSStyleSheet(options)</dfn></dt>
 	<dd>
 		When called, execute these steps:
 
-		1. Construct a new {{CSSStyleSheet}} object <var>sheet</var>,
-			with location set to the {{Document}}'s <a spec=html>base URL</a>,
+		1. Construct a new <a for="/" interface>CSSStyleSheet</a> object <var>sheet</var>,
+			with location set to the <a spec=html>base URL</a> of the {{Document}} for the window where this constructor is called on,
 			no parent CSS style sheet,
 			no owner node,
 			no owner CSS rule,
-			associated document to <var>this</var>,
 			and a title set to the {{CSSStyleSheetInit/title}} attribute of <var>options</var>.
-			Set <var>sheet’s</var> origin-clean flag.
-		2. If the {{CSSStyleSheetInit/media}} attribute of <var>options</var> is a string,
+		2.	Set <var>sheet’s</var> origin-clean flag.
+			Set <var>sheet's</var> [=constructed flag=].
+			Set <var>sheet's</var> [=constructor document=] to the {{Document}} for the window where this constructor is called on.
+		3. If the {{CSSStyleSheetInit/media}} attribute of <var>options</var> is a string,
 			<a>create a MediaList object</a> from the string
 			and assign it as <var>sheet’s</var> media.
 			Otherwise, assign a copy of the value of the attribute as <var>sheet’s</var> media.
-		3. If the {{CSSStyleSheetInit/alternate}} attribute of <var>options</var> is true,
+		4. If the {{CSSStyleSheetInit/alternate}} attribute of <var>options</var> is true,
 			set <var>sheet’s</var> alternate flag.
-		4. If the {{CSSStyleSheetInit/disabled}} attribute of <var>options</var> is true,
+		5. If the {{CSSStyleSheetInit/disabled}} attribute of <var>options</var> is true,
 			set <var>sheet’s</var> disabled flag.
-		5. <a spec=css-syntax-3>Parse a stylesheet</a> from {{text}}.
+		6. Return <var>sheet</var>.
+	</dd>
+	<dt><dfn  for=CSSStyleSheet lt="constructed flag">constructed flag</dfn></dt>
+	<dd>
+		Specified when created. Either set or unset. Unset by default.
+		Signifies whether this stylesheet is made via constructor or not, so must be set only for stylesheets that are constructed using the {{CSSStyleSheet()}} or {{CSSStyleSheet(options)}} function.
+	</dd>
+	<dt><dfn  for=CSSStyleSheet lt="disallow modification flag">disallow modification flag</dfn></dt>
+	<dd>
+		Either set or unset. Unset by default. If set, modification to the stylesheet's rules are not allowed.
+	</dd>
+	<dt><dfn  for=CSSStyleSheet lt="constructor document">constructor document</dfn></dt>
```

Comment:
What are constructor document and set of adopter documents used for? They are set, but nobody ever reads them. I was hoping we wouldn't have to maintain this information, to be honest.

Reply:
It's used in line 114 (it's a really long line so you can't see it unless you scroll to the right, oops):
> If <var>sheet</var> contains one or more <a spec=css-cascade-4>@import</a> rules and <var>sheet's</var> [=set of adopter documents=] contains any other {{Document}} than <var>sheet's</var> [=constructor document=]

We don't want to allow `replace` calls with import rules, if it's already used in another document. 
See https://github.com/WICG/construct-stylesheets/issues/57#issuecomment-432644836 and https://github.com/WICG/construct-stylesheets/issues/57#issuecomment-436509375

Reply:
I see. As @bzbarsky says, a bit annoying to have to track that. Oh well.

It seems like in the setter for adoptedStyleSheets, you need to loop over all current adopted stylesheets and remove the relevant document from that stylesheets list of adopter documents, to keep it accurate in cases like 

```js
document.adoptedStyleSheets = [s1];
document.adoptedStyleSheets = [];
```

I'd also suggest adding a line like

```html
1. Let |adopterDocument| be this {{DocumentOrShadowRoot}}, if this {{DocumentOrShadowRoot}} is a {{Document}}, or this {{DocumentOrShadowRoot}}'s [=node document=], if this {{DocumentOrShadowRoot}} is a {{ShadowRoot}}.
```

Reply:
If we want to track removal as well, we'd have to make it a multiset (or make it be a pair of <Document, count>). I guess the multiset thing is easier to have in spec, is that OK?

Reply:
Yeah, I guess it would just be a [list](https://infra.spec.whatwg.org/#lists) then (instead of a set). Just be careful when removing to only remove one.